src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,495 @@
+                  "dN5Q==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAENwDQHwyZmnYaUEp0UNqqnw0tGOGnqOMBGdds6O3+JKbmmJGT" \
+                  "n0vo7sKNNcDWDsFhJFU/RLWXmHXglxSo+yw9iQ=="
+    ldap.user('user1').add(passkey=passkey_str)
+
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    tc = "su_ldap_wrong_passkey_str_test8"
+
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/data/passkey/{tc}/{record_file}"
+        client.fs.upload(src_file, f"/tmp/{record_file}")
+
+    output = client.auth.passkey.su(tc, 'user1@test', 123456)
+    assert output.stdout_lines[-1] == 'su: Authentication failure'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su_check_when_srv_offline(client: Client, ipa: IPA):
+    """
+    :title: Check authentication of user with IPA when server is not reachable
+    :setup:
+        1. Add a IPA user with passkey_mapping.
+        2. Setup IDM client with FIDO and umockdev setup
+        3. Create the recording files when authentications is successful
+        4. Add 'pam_passkey_auth = True' under pam section of sssd.conf file.
+    :steps:
+        1. Check the umockdev-run command for authentication of the user.
+        2. Check umockdev-run that the output contains the username.
+        3. Make the server offline by updating hostname of ipa server
+        4. Check authentication of user again
+    :expectedresults:
+        1. Command returns 0.
+        2. Output contains username of the user.
+        3. Server is offline.
+        4. Authentication works, getting username in output.
+    :customerscenario: False
+    """
+    user_add = ipa.user('user1').add()
+    passkey_str = "passkey:NUZMRUXIb/W8Ij1GqwCDHSCWxt/SxWxckwtQjLYi/X6Y1qZFB+HI8WO6khzAjzsz248kHbaeAf9qfmqfCky1Jg==," +\
+                  "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEIasAa8ogjPCKXeA4KY3t0W3xBRmG+E4D+MNoRIAJrYuNLSYtAcOL7DCb" \
+                  "Ifgc+7c5Y4Mh/FzoEyeumKGYMoyTfg=="
+    user_add.passkey_add(passkey_str)
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    tc = "su_ipa_test3"
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/data/passkey/{tc}/{record_file}"
+        client.fs.upload(src_file, f"/tmp/{record_file}")
+
+    output1 = client.auth.passkey.su(tc, 'user1@ipa.test', 123456)
+
+    # Here we are making server/backend offline but not deleting cache and logs.
+
+    client.sssd.config.remove_option('domain/test', 'ipa_server')
+    client.sssd.domain["ipa_server"] = "master.ipa.server.com"
+    client.sssd.start()
+
+    output2 = client.auth.passkey.su(tc, 'user1@ipa.test', 123456)
+    assert output1.stdout_lines[-1] == output2.stdout_lines[-1] == 'user1'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_passkey__su_check_when_srv_offline_firewalld(client: Client, ldap: LDAP):
```

Comment:
I know I said earlier that you could keep the IPA and LDAP offline tests separate with the hostname and firewalld changes.  I'm wondering now as I'm reading them, if it would be possible to combine and run both unreachable hostname vs unreachable port for both LDAP and IPA in combined test cases.

What was the issue again with blocking firewall ports for IPA?   The firewall rules didn't work because a port wasn't open on the client where you ran firewalld?

Reply:
I updated what I understand, please do check.

Reply:
Marking resolved since those tests were removed from this PR.